gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/rebind/PresenterInspector.java

```diff
@@ -205,7 +208,11 @@ public String getPresenterClassName() {
      */
     public void writeProviderAssignation(SourceWriter writer) {
 
-        if (proxyStandardAnnotation != null) {
+        if (customProviderAnnotation != null) {
+            JClassType customProvider = oracle.findType(customProviderAnnotation.value().getName());
+            writer.println("presenter = new " + customProvider.getQualifiedSourceName() + "<" + presenterClassName
+                        + ">( ginjector." + getPresenterMethodName + "() );");
```

Comment:
From your javadoc snippet I believe there should not be a generic here?
Also I think I misled you with my first comments. One of `ProxyStandard`, `ProxyCodeSplit` or `ProxyCodeSplitBundle` should be mandatory because it's how the ginjector generator will determine what to generate. So I think you can revert the last 2 commits. Again I'm really sorry about this, the initial proposition was perfect.